File: components/digid_balie/config/application.rb

# Deze broncode is openbaar gemaakt vanwege een Woo-verzoek zodat deze
# gericht is op transparantie en niet op hergebruik. Hergebruik van 
# de broncode is toegestaan onder de EUPL licentie, met uitzondering 
# van broncode waarvoor een andere licentie is aangegeven.
# 
# Het archief waar dit bestand deel van uitmaakt is te vinden op:
#   https://github.com/MinBZK/woo-besluit-broncode-digid
# 
# Eventuele kwetsbaarheden kunnen worden gemeld bij het NCSC via:
#   https://www.ncsc.nl/contact/kwetsbaarheid-melden
# onder vermelding van "Logius, openbaar gemaakte broncode DigiD" 
# 
# Voor overige vragen over dit Woo-besluit kunt u mailen met open@logius.nl
# 
# This code has been disclosed in response to a request under the Dutch
# Open Government Act ("Wet open Overheid"). This implies that publication 
# is primarily driven by the need for transparence, not re-use.
# Re-use is permitted under the EUPL-license, with the exception 
# of source files that contain a different license.
# 
# The archive that this file originates from can be found at:
#   https://github.com/MinBZK/woo-besluit-broncode-digid
# 
# Security vulnerabilities may be responsibly disclosed via the Dutch NCSC:
#   https://www.ncsc.nl/contact/kwetsbaarheid-melden
# using the reference "Logius, publicly disclosed source code DigiD" 
# 
# Other questions regarding this Open Goverment Act decision may be
# directed via email to open@logius.nl

# frozen_string_literal: true

require_relative "boot"
require 'rails/all'

# Require the gems listed in Gemfile, including any gems
# you've limited to :test, :development, or :production.
Bundler.require(*Rails.groups)

file_path = File.join(__dir__, "application.yml")
APP_CONFIG = ENV['ASSET_PRECOMPILE'].blank? ? DigidUtils::Settings.load_as("digid_balie", file_path) : YAML.load_file(file_path)[Rails.env]

require_relative '../app/middleware/host_header_rewrite'

module DigidBalie
  class Application < Rails::Application
    # Settings in config/environments/* take precedence over those specified here.
    # Application configuration should go into files in config/initializers
    # -- all .rb files in that directory are automatically loaded.

    # Set Time.zone default to the specified zone and make Active Record auto-convert to this zone.
    # Run "rake -D time" for a list of tasks for finding time zone names. Default is UTC.
    config.time_zone = 'Amsterdam'

    # The default locale is :en and all translations from config/locales/*.rb,yml are auto loaded.
    # config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb,yml}').to_s]
    config.i18n.default_locale = :nl
    config.middleware.insert_before Rack::Sendfile, HostHeaderRewrite

    # Default headers
    config.action_dispatch.default_headers = {
      'Content-Security-Policy' => "default-src 'self'; img-src 'self' data:; script-src 'self' 'unsafe-inline'; style-src 'self' 'unsafe-inline'",
      'X-Frame-Options' => 'SAMEORIGIN',
      'X-XSS-Protection' => '1; mode=block',
      'X-Content-Type-Options' => 'nosniff'
    }

    # Prepend all log lines with the following tags.
    config.log_tags = [:uuid]

    if APP_CONFIG['sys_logger_enabled']
      config.logger = Syslogger.new("balie-app (#{Rails.env})", Syslog::LOG_PID, Syslog::LOG_LOCAL0)
    else
      config.logger = Logger.new("#{Rails.root}/log/logger.log")
    end

    # Exception_notifier
    config.exception_notifier = APP_CONFIG['exception_notifier']
  end
end

File: components/digid_balie/app/assets/javascripts/application.js

/*
 * Deze broncode is openbaar gemaakt vanwege een Woo-verzoek zodat deze
 * gericht is op transparantie en niet op hergebruik. Hergebruik van 
 * de broncode is toegestaan onder de EUPL licentie, met uitzondering 
 * van broncode waarvoor een andere licentie is aangegeven.
 * 
 * Het archief waar dit bestand deel van uitmaakt is te vinden op:
 *   https://github.com/MinBZK/woo-besluit-broncode-digid
 * 
 * Eventuele kwetsbaarheden kunnen worden gemeld bij het NCSC via:
 *   https://www.ncsc.nl/contact/kwetsbaarheid-melden
 * onder vermelding van "Logius, openbaar gemaakte broncode DigiD" 
 * 
 * Voor overige vragen over dit Woo-besluit kunt u mailen met open@logius.nl
 * 
 * This code has been disclosed in response to a request under the Dutch
 * Open Government Act ("Wet open Overheid"). This implies that publication 
 * is primarily driven by the need for transparence, not re-use.
 * Re-use is permitted under the EUPL-license, with the exception 
 * of source files that contain a different license.
 * 
 * The archive that this file originates from can be found at:
 *   https://github.com/MinBZK/woo-besluit-broncode-digid
 * 
 * Security vulnerabilities may be responsibly disclosed via the Dutch NCSC:
 *   https://www.ncsc.nl/contact/kwetsbaarheid-melden
 * using the reference "Logius, publicly disclosed source code DigiD" 
 * 
 * Other questions regarding this Open Goverment Act decision may be
 * directed via email to open@logius.nl
*/

//= require jquery3
//= require jquery_ujs

//= require constants
//= require print_page
//= require validator

// highlights the form item if a field receives focus
(function() {
  var init, registerFieldFocus;

  registerFieldFocus = function() {
    $('fieldset :input:visible').focus(function() {
      $(this).parent().addClass('form__item--active');
    }).blur(function() {
      $(this).parent().removeClass('form__item--active');
    });
  };

  init = function() {
    registerFieldFocus();
    Validator.init();
  };

  $('input[name="sms_received_form[sms_received]"]').change(function(event) {
    if ($(this).val() == "false") {
      $('input[type="submit"]').attr("value", "SMS opnieuw verzenden");
    } else if ($(this).val() == "true") {
      $('input[type="submit"]').attr("value", "Aanvraag afronden");
    }
  })

  $(document).ready(init);
  $(document).on('page:load', init);

}).call(this);
